robotoff/insights/annotate.py

```diff
@@ -107,12 +107,15 @@ def _annotate(
         if automatic:
             insight.automatic_processing = True
 
-        insight.save()
-
         if annotation == 1 and update:
-            return self.process_annotation(insight, data=data, auth=auth)
+            annotation_result = self.process_annotation(insight, data=data, auth=auth)
+        else:
+            annotation_result = SAVED_ANNOTATION_RESULT
 
-        return SAVED_ANNOTATION_RESULT
+        insight.annotated_result = annotation_result
```

Comment:
annotation_result is a AnnotationResult object, so it should be:

```suggestion
        insight.annotated_result = annotation_result.status
```
Although I'm not sure if using a string is best…

Ideally, we should modify AnnotationResult to a an "id" (int) field, which 
* would be set to the int representing the status (see AnnotationStatus enum). 
* For that we would need to change all AnnotationResult instanciation, to add this parameter.
* and of course we would need to use an int field